src/routeApp/controllers.js

```diff
@@ -0,0 +1,66 @@
+import { matchedData } from 'express-validator';
+import { handleGetSingle } from '../controllers/baseControllers.js';
+import handleResponse from '../controllers/handleResponse.js';
+import { findAllUsers as findAllRoutes } from '../services/baseServices.js';
+import Route from './models.js';
+import { handelDeleteRoute, handeleditRoute } from './services.js';
+import { RouteCodeGenerator } from './utils.js';
+
+export const getRoutes = async (req, res) => {
+	try {
+		const routes = await findAllRoutes(Route, res);
+		return handleResponse(res, 200, routes);
+		/* c8 ignore next 3 */
+		// eslint-disable-next-line no-empty
+	} catch (error) {}
+};
+
+export const getSingleRoute = async (req, res) => {
+	const oneRoute = await handleGetSingle(Route, req, res);
+	return handleResponse(res, 200, oneRoute);
+};
+
+export const createRoute = async (req, res) => {
+	// const existRoute = RouteOption.includes()
+	try {
+		const data = matchedData(req, {
+			optional: true,
+		});
+		data.code = RouteCodeGenerator();
+
+		const newRoute = Route.create(data);
+
+		const routeExist = await Route.findOne({
+			where: { origin: data.origin, destination: data.destination },
+		});
+		if (!routeExist) {
```

Comment:
Here you can replace thos two lines with the generic `Route.createAndSave` method for clean code. And remember to check where the route exist before you call that function